client/my-sites/marketplace/pages/marketplace-plugin-upload-status/index.tsx

```diff
@@ -62,6 +64,31 @@ const MarketplacePluginInstall = ( { productSlug } ): JSX.Element => {
 		getStatusForPlugin( state, siteId, productSlug )
 	);
 
+	const supportsAtomicUpgrade = useRef< boolean >();
```

Comment:
That's an interesting use of `useRef`, can't we use `useState` instead and pass `supportsAtomicUpgrade` as a dependency later in 
https://github.com/Automattic/wp-calypso/pull/57925/files#diff-b9d1e3ec36a8799dd188abca2d80e61b898eb69540f0fdb44742afd69ccc8e95R84-R91

Reply:
This value will be passed down through a callback, which means if we use a state we'll be passing the value down to the callback when creating it. That is not the case with the `useRef` as we pass the reference thus get an updated value when this callback is called.